Formula/helix.rb

```diff
@@ -0,0 +1,33 @@
+class Helix < Formula
+  desc "Post-modern modal text editor"
+  homepage "https://helix-editor.com"
+  url "https://github.com/helix-editor/helix.git",
+       tag:      "v0.6.0",
+       revision: "ac1b7d8d0a608f47edfee2872d414e94fd26cc31"
+  license "MPL-2.0"
+  head "https://github.com/helix-editor/helix.git", branch: "master"
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", *std_cargo_args(path: "helix-term")
+    libexec.install "runtime"
+    bin.env_script_all_files libexec/"bin", HELIX_RUNTIME: libexec/"runtime"
+  end
+
+  test do
+    require "open3"
+    Open3.popen3("#{bin}/hx") do |stdin, _stdout, stderr, wait_thr|
+      on_macos do
+        error_msg = "Piping into helix-term is currently not supported on macOS"
+        assert_match error_msg, stderr.read
+        assert_equal 1, wait_thr.value.exitstatus
+      end
+      on_linux do
+        stdin.write "test"
+        sleep 5.seconds
+        system "kill", "-SIGTERM", wait_thr.pid.to_s
```

Comment:
I took a quick look at how the vim test works, perhaps we can do something similar if homebrew supports injecting keys: https://github.com/Homebrew/homebrew-core/blob/7df9f2af66192beac0479706126901e84c147503/Formula/vim.rb#L73-L79

`i` `hello` `esc` `:wq test.txt`

Reply:
I don't think this is injecting keys. Actually in the test script of `vim` and `neovim`, they are creating a command script  (i.e., `commands.vim`) and feed it to the editor. Unfortunately I don't see such feature in Helix (based on the latest release).

> we can do something similar if homebrew supports injecting keys

I agree, and looks like sending key sequences is the only way we can control Helix in the test script.

Reply:
You can use Ruby's [`PTY` module](https://ruby-doc.org/stdlib-2.6.3/libdoc/pty/rdoc/PTY.html) if you want to inject keys.

Reply:
Thanks for the advice. I tried the following test script but `hx` still hangs. When I replace `bin/"hx"` with `"vim"`, it just works. Maybe I am not using the right key sequences to control Helix. I need some help (maybe from @archseer, who is a contributor of Helix).
```ruby
test do
  require "pty"
  require "io/console"

  test_file = testpath / "test.txt"

  r, w, pid = PTY.spawn bin/"hx", test_file
  r.winsize = [80, 43]
  sleep 1
  w.write "itest\e"
  sleep 1
  w.write ":wq\n"
  sleep 1
  Process.wait(pid)
  assert_equal "test", test_file.read.chomp
end
```

Reply:
@henryhchchc can you try enabling `r.raw!` right after spawning?

Reply:
We also expect the terminal to be a tty: https://github.com/helix-editor/helix/blob/d62ad8b595a4f901b9c5dba1bb6e8f70ece395bf/helix-term/src/application.rs#L147-L153

But that should be handled by PTY?